edunext_openedx_extensions/edunext/middleware.py

```diff
@@ -78,3 +81,68 @@ def clear_cache(sender, instance, **kwargs):  # pylint: disable=unused-argument
         """
         cache_key = "redirect_cache." + fasthash(instance.domain)
         cache.delete(cache_key)  # pylint: disable=maybe-no-member
+
+
+class PathRedirectionMiddleware(object):
+    """
+    Middleware to create custom responses based on the request path
+    """
+    LOGIN_REQUIRED = "login_required"
+    NOT_FOUND = "not_found"
+    NOT_FOUND_LOGGEDIN = "not_found_loggedin"
+    NOT_FOUND_LOGGEDOUT = "not_found_loggedout"
+    REDIRECT_ALWAYS = "redirect_always"
+    REDIRECT_LOGGEDIN = "redirect_loggedin"
+    REDIRECT_LOGGEDOUT = "redirect_loggedout"
+
+    def process_request(self, request):
+        """
+        This middleware processes the path of every request and determines if there
+        is a configured action to take.
+        """
+
+        if microsite.has_override_value("EDNX_CUSTOM_PATH_REDIRECTS"):
```

Comment:
Better if we return if the setting was not overriden? and if it was, just continue the code below?